cmd/snap-confine/snap-confine-invocation.c

```diff
@@ -144,3 +145,66 @@ void sc_check_rootfs_dir(sc_invocation *inv) {
 
     die("cannot locate base snap %s", inv->base_snap_name);
 }
+
+static char* read_homedirs_from_system_params(void)
+{
+    FILE *f SC_CLEANUP(sc_cleanup_file) = NULL;
+    f = fopen("var/lib/snapd/system-params", "r");
```

Comment:
Is it deliberate that this is a relative path? If so maybe a comment about the rational?